capstone/src/main/java/com/google/sps/servlets/DeletePollOptionServlet.java

```diff
@@ -0,0 +1,81 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+/**
+ * This servlet is called weekly to delete the top challenge suggestion so that
+ * the new weekly challenge can be updated to be the next top voted suggestion
+ * following the one that has just been deleted.
+ *
+ * @author lucyqu
+ *
+ */
+@WebServlet("delete-top-option")
+public class DeletePollOptionServlet extends HttpServlet {
+
+  // Keeps track of max number of votes an option has
+  private int maxVotes = 0;
+  // Keeps track of the option with the max number of votes
+  private long maxVotesId = 0;
```

Comment:
This concerns me having these as values stored on the servlet itself.

1. `setMaxVotesAndId` does not clear the previous values leaving it prone to issues if you have multiple different polls.
2. `deleteEntity` does not update the `maxVotesId` on call which means calling it multiple times would only delete one item.

I think the simplest solution to these problems is to just return maxVotesId from your current `setMaxVotesAndId`(rename the function), and pass that value into deleteEntity, since you don't seem to need `this.maxVotes` outside of `setMaxVotesAndId`